### PR TITLE
Purchases: Perform paths cleanup in Manage Purchases section

### PR DESCRIPTION
**Pull request by gziolo**
This PR tries to refactor Manage Purchases section to use Calypso url and sections patterns.

### Changes introduced
1. For all url starting with `/purchases/*` add `/me` prefix, i.e. https://wordpress.com/me/purchases in Calypso.
2. Rename https://wordpress.com/payment-methods/add-credit-card to https://wordpress.com/me/purchases/add-credit-card in Calypso.
3. Rename https://wordpress.com/me/billing to https://wordpress.com/me/purchases/billing.
4. Merge section chunks related to purchases, billing and payment methods.
5. Add url redirects to make sure all old urls still work.

### Testing
1. Open old `Manage Purchases` section url http://calypso.localhost:3000/purchases.
2. Make sure it redirects to http://calypso.localhost:3000/me/purchases.
3. Make sure `Manage Purchases` is highlighted in the sidebar.
4. Go to `Billing History` section.
5. You should see the following url in your browser: http://calypso.localhost:3000/me/purchases/billing.
6. Type http://calypso.localhost:3000/me/billing to make sure it redirects back to the url from (5).
7. Click on `View Receipt` link in one of the receipt rows.
8. You should receipt details.
9. Click `Back` button.
10. You should get back to `Billing History` page.
11. Click `Add Credit Card` button.
12. You should see the following url in your browser: http://calypso.localhost:3000/me/purchases/add-credit-card.
13. Type http://calypso.localhost:3000/payment-methods/add-credit-card to make sure it redirects back to the url from (12).
14. Click `Back` button.
15. You should get back to `Billing History` page.
16. Click `Go to "Purchases" to add or cancel a plan.`.
17. You should see `Manage Purchases` page again.
18. Click on any purchase and then follow navigation links and play around.

All old urls have redirects in place. Pattern is described in `Changes introduced` section.

### Server changes
I'm also working on a backend patch that will update all urls to new patterns.

### CODE REVIEW

File: client/my-sites/site-settings/controller.js [hunk 1]
```diff
@@ -179,8 +180,8 @@ module.exports = {
 				notifications: '/me/notifications',
 				disbursements: '/me/public-profile',
 				earnings: '/me/public-profile',
-				'billing-history': '/me/billing',
-				'billing-history-v2': '/me/billing',
+				'billing-history': purchasesPaths.list(),
```
Comment (gziolo) [inline client/my-sites/site-settings/controller.js]: It should be `purchasesPaths.billingHistory()`

**Comment by scruffian** [comment]
Should we also move this file: https://github.com/Automattic/wp-calypso/blob/master/assets/stylesheets/_components.scss#L175

File: client/me/purchases/paths.js [hunk 2]
```diff
@@ -1,5 +1,17 @@
 function list() {
```
Comment (scruffian) [inline client/me/purchases/paths.js]: While we are updating things, this function could have a better name. Maybe `purchasesRoot`?

**Comment by gziolo** [comment]
@scruffian We probably should move `me/billing-history/` folder to `me/purchases/billing-history`. I didn't want to touch that in this PR, because this code needs some more refactoring, but I'm okey with doing it in one batch if you are happy to review it :)

File: client/me/purchases/index.js [hunk 3]
```diff
@@ -1,16 +1,70 @@
...
+		);
+
+		// redirect legacy urls
+		page(
+			'/payment-methods/add-credit-card',
+			() => page.redirect( paths.addCreditCard() )
+		);
+	}
+
+	if ( config.isEnabled( 'me/billing-history' ) ) {
```
Comment (scruffian) [inline client/me/purchases/index.js]: I think we can probably remove this config value now.

**Comment by scruffian** [comment]
The code here looks good 👍

File: client/me/purchases/index.js (hunk 3 above)
Comment (gziolo) [inline client/me/purchases/index.js]: Good idea, I'll remove it.

File: client/me/purchases/paths.js (hunk 2 above)
Comment (gziolo) [inline client/me/purchases/paths.js]: Yes, sounds better :)

Comment (gziolo) [inline client/me/purchases/paths.js]: Fixed with 67b5c1185b3a3faf0b759d3f370e57aedca5f4c0.

File: client/me/purchases/index.js (hunk 3 above)
Comment (gziolo) [inline client/me/purchases/index.js]: Fixed with  66017afc8271c9cfc2d92f2c549d7aa616b31d3d.

**Pull request merged**